lib/llvmopencl/Workgroup.cc

```diff
@@ -1276,7 +1280,8 @@ LLVMValueRef Workgroup::createArgBufferLoad(LLVMBuilderRef Builder,
   LLVMValueRef Offs =
       LLVMConstInt(LLVMInt32TypeInContext(LLVMContext), ArgPos, 0);
   LLVMValueRef ArgByteOffset =
-      LLVMBuildGEP(Builder, ArgBufferPtr, &Offs, 1, "arg_byte_offset");
+      LLVMBuildGEP2(Builder, LLVMTypeOf(ArgBufferPtr), ArgBufferPtr, &Offs, 1,
```

Comment:
Changing the Workgroup.cc:s LLVMBuildGEP2's second argument to LLVMGetElementType(LLVMTypeOf(ArgBufferPtr)), or alternatively to LLVMInt8TypeInContext(LLVMContext) seems to fix the type mismatch assertions @pjaaskel reported.